release/serve_tests/workloads/autoscaling_multi_deployment_1k.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+"""
+Benchmark test for multi deployment with autoscaling at up to 1k no-op replica
+scale.
+
+1) Start with a single head node.
+2) Start 1000 deployments each with up to 10 no-op replicas
+3) Launch wrk in each running node to simulate load balanced request
+4) Recursively send queries to random deployments, up to depth=5
+5) Run a 10-minute wrk trial on each node, aggregate results.
+
+Report:
+    per_thread_latency_avg_ms
+    per_thread_latency_max_ms
+    per_thread_avg_tps
+    per_thread_max_tps
+
+    per_node_avg_tps
+    per_node_avg_transfer_per_sec_KB
+
+    cluster_total_thoughput
+    cluster_total_transfer_KB
+    cluster_max_P50_latency_ms
+    cluster_max_P75_latency_ms
+    cluster_max_P90_latency_ms
+    cluster_max_P99_latency_ms
+"""
+
+import click
+import math
+import os
+import random
+
+import ray
+from ray import serve
+from ray.serve.utils import logger
+from serve_test_utils import (
+    aggregate_all_metrics,
+    run_wrk_on_all_nodes,
+    save_test_results,
+)
+from serve_test_cluster_utils import (
+    setup_local_single_node_cluster,
+    setup_anyscale_cluster,
+    warm_up_one_cluster,
+    NUM_CPU_PER_NODE,
+    NUM_CONNECTIONS,
+)
+from typing import Optional
+
+# Experiment configs
+DEFAULT_SMOKE_TEST_MIN_NUM_REPLICA = 1
+DEFAULT_SMOKE_TEST_MAX_NUM_REPLICA = 4
+DEFAULT_SMOKE_TEST_NUM_DEPLOYMENTS = 4  # 1 replicas each
+
+# TODO:(jiaodong) We should investigate and change this back to 1k
+# for now, we won't get valid latency numbers from wrk at 1k replica
+# likely due to request timeout.
+DEFAULT_FULL_TEST_MIN_NUM_REPLICA = 1
+DEFAULT_FULL_TEST_MAX_NUM_REPLICA = 1000
+# TODO(simon): we should change this back to 100. But due to long poll issue
+# we temporarily downscoped this test.
+# https://github.com/ray-project/ray/pull/20270
+DEFAULT_FULL_TEST_NUM_DEPLOYMENTS = 10  # 100 replicas each
```

Comment:
Where does "100 replicas each" come from?  Is that the target num replicas for each deployment?

Reply:
Ah yes it is. There's a bug in the test, where instead of passing in 100 replicas as the `max_replicas`, it passed in 1000. I'll fix it.